lib/src/vector_tile_provider.dart

```diff
@@ -95,6 +85,36 @@ class MemoryCacheVectorTileProvider extends VectorTileProvider {
   }
 }
 
+/// [assetPathTemplate] the asset path template, e.g. `'assets/map_tiles/{z}/{x}/{y}.pbf'`
+/// [maximumZoom] the maximum zoom supported by the tile provider, not to be
+/// confused with the maximum zoom of the map widget. The map widget will
+/// automatically use vector tiles from lower zoom levels once the maximum
+/// supported by this provider is reached.
+class AssetVectorTileProvider extends VectorTileProvider {
```

Comment:
Let's have this in a separate source file.

Reply:
As in putting it somewhere like `asset_vector_tile_provider.dart`?